crates/net/network/src/transactions/fetcher.rs

```diff
@@ -1072,13 +1081,45 @@ impl Default for TransactionFetcher {
     }
 }
 
+#[derive(Debug, Clone)]
+pub(super) struct TxSizeMetadata {
+    pub peer_id: PeerId,
+    pub tx_encoded_len: usize,
+}
+
+impl PartialEq for TxSizeMetadata {
+    fn eq(&self, other: &Self) -> bool {
+        self.peer_id == other.peer_id
+    }
+}
+
+impl Eq for TxSizeMetadata {}
+
+impl Hash for TxSizeMetadata {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.peer_id.hash(state);
+    }
+}
+
+impl TxSizeMetadata {
+    fn get_size(&self) -> Option<usize> {
+        if self.tx_encoded_len == 0 {
+            None
+        } else {
+            Some(self.tx_encoded_len)
+        }
+    }
```

Comment:
any reason this doesn't just return `usize`?

Reply:
have you read the [docs](https://docs.rs/cranelift-entity/0.106.2/cranelift_entity/packed_option/index.html) for `PackedOption` type mentioned in the issue? it's similar to that type

Reply:
yes, i understand that it is the same as `PackedOption`, but i do not see why we need an option in the first place. is it because you want 0 to represent that the peer has not told us how big the tx is?